src/vlt-json/test/index.ts

```diff
@@ -337,3 +337,53 @@ t.test('do not walk up past a .git folder', async t => {
   unload()
   t.equal(load('root', assertBool), false)
 })
+
+t.test(
+  'create directories when saving to non-existent path',
+  async t => {
+    const dir = t.testdir({
+      a: {
+        b: {
+          'package.json': JSON.stringify({}),
+          c: {},
+        },
+      },
+    })
+    t.chdir(resolve(dir, 'a/b/c'))
+
+    const { load, save, find } = await t.mockImport<
+      typeof import('../src/index.ts')
+    >('../src/index.ts', {
+      'node:os': t.createMock(await import('node:os'), {
+        homedir: () => t.testdirName,
+      }),
+    })
+
+    // The find function should return a path that doesn't exist yet
+    const configPath = find()
+    t.equal(configPath, resolve(dir, 'a/b/vlt.json'))
+
+    // Directory doesn't exist initially
+    t.equal(
+      await lstat(resolve(dir, 'a/b/vlt.json')).catch(() => false),
+      false,
+    )
+
+    const assertBool = (b: unknown): asserts b is boolean => {
+      if (b !== !!b) throw new Error('expected boolean')
+    }
+
+    // Load should return undefined since file doesn't exist
+    t.equal(load('testField', assertBool), undefined)
+
+    // Save should create the directory and file
+    save('testField', true)
```

Comment:
I think we'll need to also set the which value to point to user:
```suggestion
    // Save should create the directory and file
    save('testField', true, 'user')
```
but then we might need to tweak something else in order to have the "user" point to the `a/d/` mock folder we want to use